src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/timeserved/TimeServedProbationConfirmContactService.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.timeserved
+
+import org.slf4j.LoggerFactory
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.timeserved.TimeServedProbationConfirmContact
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.timeserved.TimeServedProbationConfirmContactRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.timeserved.TimeServedProbationConfirmContactRepository
+import java.time.LocalDateTime
+
+@Service
+class TimeServedProbationConfirmContactService(
+  private val repo: TimeServedProbationConfirmContactRepository,
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(TimeServedProbationConfirmContactService::class.java)
+  }
+
+  @Transactional
+  fun addConfirmContact(licenceId: Long, request: TimeServedProbationConfirmContactRequest) {
+    val username = SecurityContextHolder.getContext().authentication.name
+
+    val existing = repo.findByLicenceId(licenceId)
+    if (existing != null) {
+      log.info("Updating existing TimeServedProbationConfirmContact for licenceId $licenceId")
+      existing.contactStatus = request.contactStatus
+      existing.communicationMethods = request.communicationMethods
+      existing.otherDetail = request.otherCommunicationDetail
+      existing.confirmedByUsername = username
+      existing.dateLastUpdated = LocalDateTime.now()
+    } else {
```

Comment:
I think we don't want to create or update - just create each time -so we have a history of previous entries.